builtins/web/fetch/request-response.cpp

```diff
@@ -1764,7 +1764,7 @@ bool Request::initialize(JSContext *cx, JS::HandleObject request, JS::HandleValu
   }
 
   // 26.  If `init["signal"]` exists, then set `signal` to it.
-  if (!signal_val.isUndefined()) {
+  if (AbortSignal::is_instance(signal_val)) {
```

Comment:
Ugh, except I mislead you: this check will successfully avoid the type safety issues we'd otherwise have, but aren't what the spec requires. What we really need is something along these lines:
```
if (!signal_val.isUndefined()) {
  if (AbortSignal::is_instance(signal_val)) {
    signal_obj = signal_val.toObjectOrNull();
  } else {
    // Throw a TypeError
  }
}

Reply:
It looks like we have to account for `signal_val == NullValue` which is defined but shouldn't error the constructor -- only overwrite the previous value. Something like this:

```
  if (!signal_val.isUndefined()) {
    signal_obj = signal_val.toObjectOrNull();
    if (signal_obj && !AbortSignal::is_instance(signal_obj)) {
      api::throw_error(cx, FetchErrors::InvalidSignal);
      return false;
    }
  }
```